courses/intro-to-ada/chapters/modular_programming.rst

```diff
@@ -237,3 +239,434 @@ This example shows how :ada:`Last_Increment` is used indirectly:
+above and implement a :ada:`Week.Child_2` package. For example:
+
+.. code:: ada no_button
+
+    package Week.Child_2 is
+
+       function Get_Last_Of_Week return String;
+
+    end Week.Child_2;
+
+Here, :ada:`Week` is still the parent package of the :ada:`Child` package,
+but it's also the parent of the :ada:`Child_2` package. In the same way,
+:ada:`Child_2` is obviously one of the child packages of :ada:`Week`.
+
+This is the corresponding package body of :ada:`Week.Child_2`:
+
+.. code:: ada no_button
+
+    package body Week.Child_2 is
+
+       function Get_Last_Of_Week return String is
+       begin
+          return Sun;
+       end Get_Last_Of_Week;
+
+    end Week.Child_2;
+
+We can now reference both children in our test application:
+
+.. code:: ada
+    :class: ada-run
+
+    with Ada.Text_IO;  use Ada.Text_IO;
+    with Week.Child;   use Week.Child;
+    with Week.Child_2; use Week.Child_2;
+
+    procedure Main is
+    begin
+       Put_Line ("First day of the week is " & Get_First_Of_Week);
+       Put_Line ("Last day of the week is "  & Get_Last_Of_Week);
+    end Main;
+
+:code-config:`reset_accumulator=True`
+
+Nested packages
+~~~~~~~~~~~~~~~
+
+Nested packages, as the name suggests, are declared within a parent
+package. This contrasts with child packages, which are declared separately.
+For example, this would be a nested package for the :ada:`Week` package:
+
+.. code:: ada no_button
+
+    package Week is
+
+       Mon : constant String := "Monday";
+       Tue : constant String := "Tuesday";
+       Wed : constant String := "Wednesday";
+       Thu : constant String := "Thursday";
+       Fri : constant String := "Friday";
+       Sat : constant String := "Saturday";
+       Sun : constant String := "Sunday";
+
+       package Nested is
+
+          function Get_First_Of_Week return String;
+
+       end Nested;
+
+    end Week;
+
+In contrast to child packages, we don't write
+:ada:`package body Week.Nested is ...` to implement the package body of
+:ada:`Nested`. Instead, the package body of :ada:`Nested` is *nested* in
+the package body of the parent package :ada:`Week`:
+
+.. code:: ada no_button
+
+    package body Week is
+
+       package body Nested is
+
+          function Get_First_Of_Week return String is
+          begin
+             return Mon;
+          end Get_First_Of_Week;
+
+       end Nested;
+
+    end Week;
+
+We can now use elements from :ada:`Week.Nested` in a test application:
+
+.. code:: ada
+    :class: ada-run
+
+    with Ada.Text_IO; use Ada.Text_IO;
+    with Week;
+
+    procedure Main is
+       use Week.Nested;
+    begin
+       Put_Line ("First day of the week is " & Get_First_Of_Week);
+    end Main;
+
+Note that we cannot access the :ada:`Week.Nested` package directly using
+:ada:`with Week.Nested` because :ada:`Nested` is actually part of :ada:`Week`,
+not  a separate subunit. We can, however, still write :ada:`use Week.Nested`
```

Comment:
1) extra space between "not" and "a" (not sure if it matters :)
2) "separate" is an Ada word - I would say  "an independent subunit"